server/platform/shared/mfa/mfa.go

```diff
@@ -120,18 +123,32 @@ func (m *MFA) Deactivate(userId string) error {
 }
 
 // Validate the provide token using the secret provided
-func (m *MFA) ValidateToken(secret, token string) (bool, error) {
+func (m *MFA) ValidateToken(user *model.User, token string) (bool, error) {
+	usedTs, err := m.store.GetMfaUsedTimestamps(user.Id)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to retrieve the DisallowReuse slice")
+	}
+
 	otpConfig := &dgoogauth.OTPConfig{
-		Secret:      secret,
-		WindowSize:  3,
-		HotpCounter: 0,
+		Secret:        user.MfaSecret,
+		WindowSize:    3,
+		HotpCounter:   0,
+		DisallowReuse: usedTs,
 	}
 
 	trimmedToken := strings.TrimSpace(token)
 	ok, err := otpConfig.Authenticate(trimmedToken)
 	if err != nil {
 		return false, errors.Wrap(err, "unable to parse the token")
 	}
+	if !ok {
+		return false, nil
+	}
+
+	err = m.store.StoreMfaUsedTimestamps(user.Id, otpConfig.DisallowReuse)
```

Comment:
Quick question, after calling `.Authenticate`, the `DisallowReuse` property contains only those timestamps inside of the window and the rest are cleared, right?

Reply:
Discussed offline, the `DisallowReuse` property is updated as part of the call